modules/consentManagement.js

```diff
@@ -0,0 +1,171 @@
+/**
+ * This module adds GDPR consentManagement support to prebid.js.  It interacts with
+ * supported CMPs (Consent Management Platforms) to grab the user's consent information
+ * and make it available for any GDPR supported adapters to read/pass this information to
+ * their system.
+ */
+import * as utils from 'src/utils';
+import { config } from 'src/config';
+import { setTimeout } from 'core-js/library/web/timers';
+
+const DEFAULT_CMP = 'appnexus';
+const DEFAULT_CONSENT_TIMEOUT = 5000;
+const DEFAULT_FAILURE_CHOICE = 'proceed';
+
+export let userCMP;
+export let consentTimeout;
+export let lookupFailureChoice;
+
+let consentId = '';
+let adUnits;
+let context;
+let args;
+let nextFn;
+
+let cmpActive;
+let timer;
+
+// add new CMPs here, with their dedicated lookup function that passes the consentString to postLookup()
+const cmpCallMap = {
+  'appnexus': lookupAppNexusConsent
+};
+
+// AppNexus CMP lookup process
+function lookupAppNexusConsent() {
```

Comment:
So this is a public API we're making here for others that want to implement CMP management which means we have to be very exact in the way its implemented.  Right now it looks very... clunky?

I think implementing the map is fine and then looking up the CMP implementation is also fine; but the function itself should probably be a [pure function](https://en.wikipedia.org/wiki/Pure_function) for a few reasons: Firstly, if we expect this to be implemented for many different vendors then it's going to need to be thoroughly tested and the testing setup for a function that requires so much context through closure variables is very difficult.  Second, this code could possibly be moved to further modules in the future (if we don't want to include _every_ third-party CMP implementation when there are many), but the function cannot be moved to another file when it depends on so much closure context.  Third, the API is very unclear for third-party implementors; it appears to rely on various closure function references such as `postLookup` on success or `pcmTimedOut` on fail; these should probably be passed into the function as arguments and given names that represent what they are going to do, not the condition that created them e.g. `success(consentString)` or `error()` on fail (just examples, probably better names).  Also, I don't think the third-party function should be handling the timed out case..  the platform should handle that to prevent each implementation rewriting possibly buggy but expected time out handling code.

Anyways, I think a lot more thought needs to be put into how this API is exposed and implemented.